lib/coverband/configuration.rb

```diff
@@ -55,19 +60,38 @@ def s3_secret_access_key
     end
 
     def store
-      return @store if @store
-      raise 'no valid store configured'
+      @store ||= Coverband::Adapters::RedisStore.new(coverband_redis, redis_store_options)
     end
 
     def store=(store)
       if store.is_a?(Coverband::Adapters::Base)
         @store = store
       elsif defined?(Redis) && redis && redis.is_a?(Redis)
-        @store = Coverband::Adapters::RedisStore.new(redis, ttl: Coverband.configuration.redis_ttl,
-                                                            redis_namespace: Coverband.configuration.redis_namespace)
+        @store = Coverband::Adapters::RedisStore.new(redis, redis_store_options)
       elsif store.is_a?(String)
         @store = Coverband::Adapters::FileStore.new(store)
       end
     end
+
+    private
+
+    def coverband_redis
+      # Redis doesn't default to correct local host if you pass url: nil
+      redis_url ? Redis.new(url: redis_url) : Redis.new
```

Comment:
Not that important but Redis.new(url: nil) seems to work for me:
```
[4] pry(main)> Redis.new(url: nil)
#<Redis client v4.0.3 for redis://127.0.0.1:6379/0>
[5] pry(main)>
```

Reply:
weird I swear I had had an issue with that in the past...

Reply:
fixed